hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/FSQueue.java

```diff
@@ -242,7 +244,9 @@ public void setFairSharePreemptionThreshold(float fairSharePreemptionThreshold)
    * @return true if the queue can be preempted
    */
   public boolean canBePreempted() {
-    assert parent != null;
+    Preconditions.checkNotNull(parent, "Parent queue can't be null since"
```

Comment:
Maybe, we could make this message more clear. "Parent queue is null. Looks like we are checking if root can be preempted."

Alternatively, can we make the if check (parent != null && ...)? That way, else would capture the null case and things should work fine?